app/components/spin-button.js

```diff
@@ -90,16 +91,19 @@ export default Ember.Component.extend({
       this.setDisabled();
 
       if (this.get('startDelay') > 4) {
-        Ember.run.later(this, this.createSpinner, element, this.get('startDelay'));
+       this.set('_spinnerTimer', Ember.run.later(this, this.createSpinner, element, this.get('startDelay')));
       }else{
         this.createSpinner(element);
       }
     }else{
+      Ember.run.cancel(this.get('_spinnerTimer'));
+      this.set('_spinnerTimer', null);
       this.setEnabled();
     }
   }.observes('inFlight'),
 
   createSpinner: function(element) {
+    this.set('inFlightStyle', true);
```

Comment:
Well, I don't see the way out - there must be a breaking change made anyway... The thing is that  a button can be in 2 different states: 
1 . disabled
2. disabled and with a spinner

The easiest way to differentiate between these states is to add another property - in my case I called it `_showInFlightAnimation` (in my latest push I added underscore before the name to make it look "private"). Yes, internally there are two properties now, but they don't really do the same thing - one is about the public state of the button and it can be used to control the button, the other one is about the way the button is rendered.

I also don't want to change the meaning or responsibility of `inFlight` property - it designates when the button is "working".

I don't know man, maybe I still don't fully understand, what you meant by "breaking change", because more I look on it, more I think that it is just internal representation of the addon that has changed - public behaviour and public properties stayed intact...